pcdsdevices/sxr_test_absorber.py

```diff
@@ -1,39 +1,51 @@
 """
 Module for the SXR Test Absorbers.
 """
-from lightpath import LightpathState
 from ophyd.device import Component as Cpt
 
 from .epics_motor import BeckhoffAxisNoOffset
-from .interface import BaseInterface, LightpathMixin
+from .inout import TwinCATInOutPositioner
+from .interface import BaseInterface, LightpathInOutCptMixin
+from .signal import PytmcSignal
 
 
-class SxrTestAbsorber(BaseInterface, LightpathMixin):
+class ST3K4AutoError(RuntimeError):
+    ...
+
+
+class SxrTestAbsorberStates(TwinCATInOutPositioner):
+    """
+    SXR Test absorber states
+
+    This has some automation where it tracks the position of ST3K4.
+    Accordingly, it has a configuration parameter for enabling/disabling this.
+    """
+    st3k4_auto = Cpt(PytmcSignal, ':ST3K4_AUTO', io='io', kind='config')
+
+    def check_value(self, pos):
+        rval = super().check_value(pos)
+        if self.st3k4_auto.get():
+            raise ST3K4AutoError(
+                "ST1K4 must follow ST3K4. Move rejected."
+            )
+        return rval
+
+
+class SxrTestAbsorber(BaseInterface, LightpathInOutCptMixin):
     """
     SXR Test Absorber: Used for testing the sxr beamline at high pulse rates.
 
-    This device has 1 main members: the stopper/absorber (diamond stopper).
+    This device has 1 main member: the stopper/absorber (diamond stopper).
 
     The vertical motor moves stopper in and out of the beam path in
     the +/- y direction.
+
+    It has a state selector that can be used to move in and out by state name.
     """
 
     tab_component_names = True
 
+    state = Cpt(SxrTestAbsorberStates, ':MMS:STATE', kind='hinted')
     absorber_vert = Cpt(BeckhoffAxisNoOffset, ':MMS:01', kind='normal')
 
-    lightpath_cpts = ['absorber_vert.user_readback']
-
-    def calc_lightpath_state(self, absorber_vert: float) -> LightpathState:
-        pos = absorber_vert
-        # 0 is out, negative is in
-        # this device has never been inserted, so we don't know the in pos
-        self._inserted = pos < -1
-        self._removed = pos > -1
-        self._transmission = int(self._removed)
-
-        return LightpathState(
-            inserted=self._inserted,
-            removed=self._removed,
-            output={self.output_branches[0]: self._transmission}
-        )
+    lightpath_cpts = ['state']
```

Comment:
Do we need to set transmission values or anything in the in/out positioner for lightpath? The requirements for that interface are a bit hazy to me

Reply:
Since this inherits from `LightpathInOutCptMixin`, it gathers information from its `lightpath_cpts`, so as long as the the `SxrTetsAbsorberStates` reports its lightpath state we should be good.  

What was removed here was hand-written behavior.  It should be easy enough to test by calling `.get_lightpath_state()` after instantiating it.  Is the IOC for this running?

Reply:
Right I definitely need to verify the lightpath bits before merging

Reply:
The lightpath bits do work as expected (provided you instantiate with happi md with input/output lines)

Reply:
```
In [1]: from pcdsdevices.sxr_test_absorber import SxrTestAbsorber
In [2]: st1k4 = SxrTestAbsorber('ST1K4:TEST', name='st1k4')
In [24]: st1k4.output_branches = ['K4']
In [25]: st1k4.input_branches = ['K4']
In [33]: st1k4._init_summary_signal()
In [34]: st1k4.get_lightpath_state()
Out[34]: LightpathState(inserted=True, removed=False, output={'K4': 0})
```
Omitted: me trying to figure out why it wasn't working